NOTICE.txt

```diff
@@ -20,5 +20,8 @@ from the Apache Spark project (www.github.com/apache/spark)
 Apache Spark
 Copyright 2014 and onwards The Apache Software Foundation.
 
+Apache Iceberg
+Copyright 2017-2024 The Apache Software Foundation
```

Comment:
Same as above, I think this change to the notice is correct, currently the kernel module doesn't have its own License/Notice so I added this to the root level.